### PR TITLE
Private events

### PR DESCRIPTION
**Pull request by tmampa**
Build an event web app that uses active association

- Events are associated with users
## Changes made
- Screenshot visible on  Readme file
- When a user clicks attend button, a list appears with all the users attending the event including the user.

### CODE REVIEW

**Review by george-shammar — CHANGES_REQUESTED** [review]
### **STATUS: INVALID** :red_circle: :red_circle: 

@tmampa , this submission is being invalidated because github actions for linters has not been set up as well as the wrong gitflow.

Please set up linters correctly and activate github actions. Also, make the development branch the default branch.

Please submit another CRR when you're ready.

Cheers :smile: With much respect. :tophat:

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_ Changes Requested :recycle:

## Hello Tshephang Mampa


## To Highlight :tada:

1. Your PR has a good title and description.
2. You have descriptive git commit messages.
3. You have mastered Git flow.


However, there are some changes that you need to make in order to get your project approved

- [ ] Kindly add a screenshot to your readme file

- [ ] I got this error when I try to create new event before logging in
err


- [ ] Kindly don't use ```return``` explicitly in your functions


- [ ] After clicking on the attend button the button should disappear or show ```attended```

- [ ] User model: has_many :events, foreign_key: :creator_id [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png)

- [ ] Event model belongs_to :creator, class_name: "User" [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png)

  .Migrations

- [ ] Events table has creator_id as the foreign key [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png)

- [ ] please use proper flags for each resources in the routes file, for example:
      resources :attended_events, only: %i[index]

- [ ] Add integration tests for the event managment features (Rspec + Capybara)[screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png)

<hr>

Please, make the REQUIRED changes and submit your request again! :grinning:
Good luck. 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

File: app/controllers/attended_events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def show
+    event = Event.find(params[:event_id])
+    @attend = AttendedEvent.where('event_id = ?', event.id)
+  end
+
+  def create
+    event = Event.find(params[:event_id])
+    unless user_signed_in?
+      redirect_to new_user_session_path
+      return
```
Comment (MkrtichSargsyan) [inline app/controllers/attended_events_controller.rb]: kindly don't use ```return``` explicitly

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
+  devise_for :users
+  resources :events
+  root 'events#index'
+  get '/users', to: 'users#show'
+  resources :attended_events
```
Comment (MkrtichSargsyan) [inline config/routes.rb]: please use proper flags for each resources, for example:
      resources :attended_events, only: %i[index]

File: app/controllers/event_attendees_controller.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    event = Event.find(params[:event_id])
+    attended = event.event_attendees.find_by(attendee_id: current_user.id)
+    redirect_to event_path(event) and return if attended
+
+    @event_attendee = event.event_attendees.build(attendee_id: current_user.id)
+    redirect_to event_path(event) and return if @event_attendee.save
+
+    redirect_to event_path(event)
+  end
+end
```
Comment (george-shammar) [inline app/controllers/event_attendees_controller.rb:12]: - [ ] As mentioned in the previous review, please avoid an explicit declaration of the "return" keyword. This is ruby convention.

**Review by george-shammar — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUIRED** :recycle: :recycle: :construction: 

Thank you for implementing some of the required changes from the previous review.
:heavy_check_mark: The error when trying to create an event before signing up has been fixed.
:heavy_check_mark: The "Creator" and relevant associations have also been implemented.
:heavy_check_mark: Plus, your repo looks professional.

Please see below list of pending required changes from the last review:

- [ ] Kindly add a screenshot to your readme file.
- [ ] Kindly don't use return explicitly in your functions. Please see my in-line comments.
- [ ] After clicking on the attend button, the button should disappear or show "attended" or "attending".
- [ ]  Please implement integration tests for the event management features (Rspec + Capybara).

Please make these required changes and submit another CRR when you're ready.

Kindly drop your message here and ping me on slack if you require any support or need clarifications.

**Cheers :smile: With respect :tophat:**

File: app/controllers/event_attendees_controller.rb (hunk 3 above)
Comment (tmampa) [reply app/controllers/event_attendees_controller.rb:12]: @george-shammar  What's the right convention to use in this case?

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle: :clap: 
---------------
Hello @tmampa, great job working on this project and in the required changes. :muscle: 
Your project is working really nice. :+1: :100: 
---------------
Now you can merge this branch :1st_place_medal:
HAPPY CODING!!!! :smile: 

well-1

**Pull request merged**